okhttp-ws/src/main/java/com/squareup/okhttp/internal/ws/RealWebSocket.java

```diff
@@ -46,8 +47,9 @@
   private volatile boolean writerSentClose;
```

Comment:
Writes are single threaded and serial. If the calling code calls close,
there's no reason for it to interact with the WebSocket again.

Even if writes are queued, a single thread will be processing them and it
will know whether it ever called close or not.

On Sat, Nov 14, 2015, 9:45 AM Jesse Wilson <notifications@github.com> wrote:

> In
> okhttp-ws/src/main/java/com/squareup/okhttp/internal/ws/RealWebSocket.java
> <https://github.com/square/okhttp/pull/1996#discussion_r44859305>:
>
> > @@ -46,8 +47,9 @@
> >    private volatile boolean writerSentClose;
>
> Do multiple threads share a single client socket socket? My guess is that
> the UI will enqueue messages to be sent asynchronously, so a single write
> that fails shouldn't cause other writes to be treated as programming errors?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/okhttp/pull/1996/files#r44859305>.
>